src/Quartz/Impl/JobDetailImpl.cs

```diff
@@ -67,13 +95,13 @@ internal class JobTypeInformation
     [Serializable]
     public class JobDetailImpl : IJobDetail
     {
-        private static readonly ConcurrentDictionary<Type, JobTypeInformation> jobTypeCache = new ConcurrentDictionary<Type, JobTypeInformation>();
-
         private string name = null!;
         private string group = SchedulerConstants.DefaultGroup;
         private string? description;
         private JobDataMap jobDataMap = null!;
         private Type jobType = null!;
```

Comment:
We actually lie about this not being nullable. We have a use case in [`StdAdoDelegate.SelectJobForTrigger(...)`](https://github.com/quartznet/quartznet/blob/73d385526dec96b4c36e03a0ef75a4dfc5a32306/src/Quartz/Impl/AdoJobStore/StdAdoDelegate.Jobs.cs#L208) where we allow an  **IJobDetail** to be built without a job type. We invoke this method with **loadJobType** set to **false** from [`JobStoreSupport.RemoveTrigger(ConnectionAndTransactionHolder conn, TriggerKey triggerKey, IJobDetail? job, CancellationToken cancellationToken = default)`](https://github.com/quartznet/quartznet/blob/73d385526dec96b4c36e03a0ef75a4dfc5a32306/src/Quartz/Impl/AdoJobStore/JobStoreSupport.cs#L1295) if the **job** argument is **null**.

One way of dealing with this would be to:
1.  Always build a full **IJobDetail** instance (wih job type). This would be the easiest solution, but I'm pretty sure this is not what you want.
2. Introduce a specialized **IJobDetail**-like type that only represents a **JobType** and whether it's durable or not. We can then remove the **loadJobType** argument from `SelectJobForTrigger(ConnectionAndTransactionHolder conn,             TriggerKey triggerKey, ITypeLoadHelper loadHelper, bool loadJobType, CancellationToken cancellationToken = default)` and introduce another (similar) method that would return the new **IJobDetail**-like type.
3. Update **JobDetailImpl.JobType** (and other properties or methods in **JobDetailImpl** that use **JobType**) to throw an exception if **JobType** was not set.  There are a few problematic methods though:
     * JobDetailImpl.ToString()
    Here we use the **JobType** property, but if this can now throw we end up with a `.ToString()` override that can 
    throw. I'm sure we do not want this.
     * JobDetailImpl.Validate()
    He were use the **jobType** (and other) instance fields directly. This is not correct from an extensibility point-of- 
    view as we allow the corresponding properties to be overridden, but ignore them.

Personally, I prefer option **2**.
Option **3** is only an option if we stop marking some properties virtual.
There are lots of places where we mark properties or methods virtual, but - since we do not always invoke these properties or methods - it does not make much sense to do so.

Reply:
There's probably a more larger concern where we would not want to rely on `Type`, when a web interface interacts and handles jobs it shouldn't be required it to have the DLL (concrete types) deployed. So a big design mistake in the early days was to require a *resolvable* type in many places, when it should have been enough to resolve it at store time.